### PR TITLE
Scheduler profile configuration and run e2e

### PR DESCRIPTION
**Pull request by AdityaHonkalas**
Signed-off-by: Aditya Honkalas <aditya.honkalas@ibm.com>

### CODE REVIEW

File: playbooks/roles/ocp-scheduler/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  block:
+    - fail:
+        msg: "Please set the environment variables GITHUB_USERNAME and GITHUB_ACCESS_TOKEN"
+      when: github_username == "" and github_personal_access_token == ""
+
+    - name: Clone the git repo
+      git:
+        repo: "https://{{ github_username }}:{{ github_personal_access_token }}@github.com/{{ scheduler_e2e_github_repo | urlsplit('path') }}"
+        dest: "{{ scheduler_workdir }}/openshift-tests-private"
+        version: "master"
```
Comment (yussufsh) [inline playbooks/roles/ocp-scheduler/tasks/main.yml:105]: I see this is hardcoded and you are not using `scheduler_e2e_github_branch` variable anywhere.

File: examples/all.yaml [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
 
+## ocp-scheduler vars
+scheduler_role_enable: false
+configure_scheduler_profile: false
+scheduler_profile: "HighNodeUtilization"
+scheduler_e2e_test_run: false
+scheduler_workdir: ~/ocp-scheduler
+scheduler_subdir_list: ["script", "e2e-test"]
+scheduler_e2e_github_repo: "https://github.com/openshift/openshift-tests-private"
+scheduler_e2e_github_branch: "master"
```
Comment (yussufsh) [inline examples/all.yaml:11]: not used anywhere

File: playbooks/roles/ocp-scheduler/README.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- The cluster is in a known good state, without any errors.
+
+
+Role Variables
+--------------
+
+| Variable                       | Required | Default                                                 | Comments                                                                           |
+|--------------------------------|----------|---------------------------------------------------------|------------------------------------------------------------------------------------|
+| scheduler_role_enable          | no       | false                                                   | Set the flag `true` to run this playbook.                                          |
+| configure_scheduler_profile    | no       | false                                                   | Set the flag `true` to configure given scheduling profile.                         |
```
Comment (yussufsh) [inline playbooks/roles/ocp-scheduler/README.md:21]: There is no default value set in the code. So this becomes Required. I feel condition will fail with None value if variable is not defined.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: playbooks/roles/ocp-scheduler/tasks/main.yml (hunk 1 above)
Comment (AdityaHonkalas) [reply playbooks/roles/ocp-scheduler/tasks/main.yml]: Used the var `scheduler_e2e_github_branch` for assigning a branch name.

File: examples/all.yaml (hunk 2 above)
Comment (AdityaHonkalas) [reply examples/all.yaml:11]: Used in the latest changes to specify git branch.

File: playbooks/roles/ocp-scheduler/README.md (hunk 3 above)
Comment (AdityaHonkalas) [reply playbooks/roles/ocp-scheduler/README.md:21]: Added in the these role variables to `roles/ocp-scheduler/defaults/main.yml` .

**Review by yussufsh — APPROVED** [review]
/approve

**Pull request merged**